packages/xpetra/src/BlockedMultiVector/Xpetra_BlockedMultiVector.hpp

```diff
@@ -742,6 +742,7 @@ namespace Xpetra {
       XPETRA_MONITOR("BlockedMultiVector::getMultiVector(r,bThyraMode)");
       TEUCHOS_TEST_FOR_EXCEPTION(r > map_->getNumMaps(), std::out_of_range, "Error, r = " << r << " is too big. The BlockedMultiVector only contains " << map_->getNumMaps() << " partial blocks.");
       XPETRA_TEST_FOR_EXCEPTION(map_->getThyraMode() != bThyraMode, Xpetra::Exceptions::RuntimeError, "BlockedMultiVector::getMultiVector: inconsistent Thyra mode");
+      (void)bThyraMode; // avoid unused parameter warning when HAVE_XPETRA_DEBUG isn't defined
```

Comment:
Note that I'm not thrilled about this solution here, but the `#ifdef` that hides the usage of `bThyraMode` is buried within `XPETRA_TEST_FOR_EXCEPTION`, so I'm not sure what else to do.